opennms-config/src/main/java/org/opennms/netmgt/config/SnmpPeerFactory.java

```diff
@@ -77,83 +77,80 @@
  */
 public class SnmpPeerFactory implements SnmpAgentConfigFactory {
     private static final Logger LOG = LoggerFactory.getLogger(SnmpPeerFactory.class);
-    private static final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
-    private static final Lock m_readLock = m_globalLock.updateLock();
-    private static final Lock m_writeLock = m_globalLock.writeLock();
+
+    private static final int VERSION_UNSPECIFIED = -1;
+
+    private static File s_configFile;
 
     /**
      * The singleton instance of this factory
      */
-    private static SnmpPeerFactory m_singleton = null;
-
-    private static File m_configFile;
+    private static SnmpPeerFactory s_singleton = null;
 
     /**
-     * The config class loaded from the config file
+     * This member is set to true if the configuration file has been loaded.
      */
-    private SnmpConfig m_config;
+    private static volatile boolean s_loaded = false;
 
-    private static FileReloadContainer<SnmpConfig> m_container;
-    private static FileReloadCallback<SnmpConfig> m_callback;
+    private final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
```

Comment:
SnmpPeerFactory was causing test failures in unrelated downstream tests because state was being held in the m_container/m_callback static fields pointing at stale configuration data.

I moved as much state as I could out of static fields and into the object instance, keeping only the singleton and loaded fields as static like we have in most of the other configuration factories. I left configFile a singleton because I wasn't exactly sure how it was used and didn't want to find out at this time.

The read/write locks are still there protecting read and writes, but they're on the object instance. setInstance/setFile are the two methods that can cause the class in the singleton replaced, and by my searching, they're only called in tests.